File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/Querying/DQ/HowToUseNotOperator.cs
using System.Collections.Generic;
using System.Linq;
using Raven.Client.Documents;
using Raven.Documentation.Samples.Orders;

namespace Raven.Documentation.Samples.ClientApi.Session.Querying.DQ
{
    public class HowToUseNotOperator
    {
        public HowToUseNotOperator()
        {
            using (var store = new DocumentStore())
            {
                using (var session = store.OpenSession())
                {
                    #region not_1
                    // load all entities from 'Employees' collection
                    // where FirstName NOT equals 'Robert'
                    List<Employee> employees = session
                        .Advanced
                        .DocumentQuery<Employee>()
                        .Not
                        .WhereEquals(x => x.FirstName, "Robert")
                        .ToList();
                    #endregion
                }

                using (var session = store.OpenSession())
                {
                    #region not_2
                    // load all entities from 'Employees' collection
                    // where FirstName NOT equals 'Robert'
                    // and LastName NOT equals 'King'
                    List<Employee> employees = session
                        .Advanced
                        .DocumentQuery<Employee>()
                        .Not
                        .OpenSubclause()
                        .WhereEquals(x => x.FirstName, "Robert")
                        .AndAlso()
                        .WhereEquals(x => x.LastName, "King")
                        .CloseSubclause()
                        .ToList();
                    #endregion
                }

                using (var session = store.OpenSession())
                {
                    #region not_3
                    // load all entities from 'Employees' collection
                    // where FirstName NOT equals 'Robert'
                    // and LastName NOT equals 'King'
                    // identical to 'Example II' but 'WhereNotEquals' is used
                    List<Employee> employees = session
                        .Advanced
                        .DocumentQuery<Employee>()
                        .WhereNotEquals(x => x.FirstName, "Robert")
                        .AndAlso()
                        .WhereNotEquals(x => x.LastName, "King")
                        .ToList();
                    #endregion
                }
            }
        }
    }
}

File: Documentation/4.0/Samples/nodejs/clientApi/session/querying/documentQuery/howToUseNotOperator.js
import { DocumentStore } from "ravendb";

const documentStore = new DocumentStore();
const session = documentStore.openSession();

async function examples() {
    {
        //region not_1
        // load all entities from 'Employees' collection
        // where firstName NOT equals 'Robert'
        const employees = await session.advanced
            .documentQuery({ collection: "Employees" })
            .not()
            .whereEquals("FirstName", "Robert")
            .all();
        //endregion
    }

    {
        //region not_2
        // load all entities from 'Employees' collection
        // where firstName NOT equals 'Robert'
        // and lastName NOT equals 'King'
        const employees = await session.advanced
            .documentQuery({ collection: "Employees" })
            .not()
            .openSubclause()
            .whereEquals("FirstName", "Robert")
            .andAlso()
            .whereEquals("LastName", "King")
            .closeSubclause()
            .all();
        //endregion
    }

    {
        //region not_3
        // load all entities from 'Employees' collection
        // where firstName NOT equals 'Robert'
        // and lastName NOT equals 'King'
        // identical to 'Example II' but 'whereNotEquals' is used
        const employees = await session.advanced
            .documentQuery({ collection: "Employees" })
            .whereNotEquals("FirstName", "Robert")
            .andAlso()
            .whereNotEquals("LastName", "King")
            .all();
        //endregion
    }
}
